community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/sort_nodes.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python3
+
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This script sorts nodes based on their `physicalHost`.
+
+See https://cloud.google.com/compute/docs/instances/use-compact-placement-policies
+
+You can reduce latency in tightly coupled HPC workloads (including distributed ML training) 
+by deploying them to machines that are located close together. 
+For example, if you deploy your workload on a single physical rack, you can expect lower latency 
+than if your workload is spread across multiple racks. 
+Sending data across multiple rack requires sending data through additional network switches. 
+
+Example usage:
+``` my_sbatch.sh
+#SBATCH --ntasks-per-node=8
+#SBATCH --nodes=64
+
+export SLURM_HOSTFILE=$(/slurm/scripts/sort_nodes.py)
+
+srun -l hostname | sort
+```
+"""
+import os
+import subprocess
+import uuid
+from typing import List, Optional, Dict, Callable
+
+def order(paths: List[List[str]]) -> List[str]:
+    """
+    Orders the leafs of the tree in a way that minimizes the sum of distance in between 
+    each pair of neighboring nodes in the resulting order.
+    The resulting order will always start from the first node in the input list.
+    The ordering is "stable" with respect to the input order of the leafs i.e.
+    given a choice between two nodes (identical in other ways) it will select "nodelist-smallest" one.
+    """
```

Comment:
From description and signature, the interface is not clear to me. It took some digging to understand that it is returning a list of node names (leaves) and that the node names are expected to be the last element in a path, I had been thinking strictly physical_host.

Something like the following would be helpful:
```
return: a list containing each leaf of the input paths, ordered as described above.  
```